### ISSUE TITLE
Incorrect PROTOBUF version

### ISSUE THREAD

**Issue by akrymski** [pytorch/pytorch#11009]
Trying to compile from source on Mac OS all goes well until:

```
/Users/akrymski/dev/pytorch/torch/lib/tmp_install/include/onnx/onnx_onnx_torch.pb.h:17:2: error: This file was generated by an older version of protoc which is
#error This file was generated by an older version of protoc which is
 ^
/Users/akrymski/dev/pytorch/torch/lib/tmp_install/include/onnx/onnx_onnx_torch.pb.h:18:2: error: incompatible with your Protocol Buffer headers. Please
#error incompatible with your Protocol Buffer headers.  Please
 ^
/Users/akrymski/dev/pytorch/torch/lib/tmp_install/include/onnx/onnx_onnx_torch.pb.h:19:2: error: regenerate this file with a newer version of protoc.
```

What protobuf version is required?  My protoc reports v3.6.0 and conda reports libprotobuf at v3.6.0 as well

**Comment by ezyang** [comment]
Do you have a copy of protobuf installed by brew in `/usr/local/include`? If so, if you could unlink it and then build again, I think that will help. Otherwise, can you get us the full command line which was used to build the file that failed?

**Comment by marcjoos-phd** [comment]
I have a related issue trying to build PyTorch 1.4.0 with gcc 8.3.0 on RHEL 7.6. Here is what I do:

- `git clone --recursive https://github.com/pytorch/pytorch`
- `cd pytorch; git checkout origin/v1.4.0`
- `USE_BLAS=MKL setup.py bdist_wheel` (that fails for some reason because it is unable to efficiently detect my Intel MKL install)
- `cd build`
- `[cmake command line from the previous setup.py] -DINTEL_MKL_DIR=$MKLROOT`
- `cd ..; USE_BLAS=MKL setup.py bdist_wheel`

I checked the protobuf version provided with PyTorch, that is currently 3.6.1. I got protobuf already installed in my Python distribution that was 3.9.0. I therefore downgraded my local protobuf installation to be in accordance with what is in PyTorch third parties... But I still get the issue.
